connectivity/netsocket/include/netsocket/NetworkStack.h

```diff
@@ -395,6 +395,51 @@ class NetworkStack : public DNS {
     virtual nsapi_size_or_error_t socket_recvfrom(nsapi_socket_t handle, SocketAddress *address,
                                                   void *buffer, nsapi_size_t size) = 0;
 
+    /** Send a packet with ancillary data over a UDP socket
+     *
+     *  Sends data to the specified address. Returns the number of bytes
+     *  sent from the buffer.
+     *
+     *  This call is non-blocking. If sendto would block,
+     *  NSAPI_ERROR_WOULD_BLOCK is returned immediately.
+     *
+     *  @param handle   Socket handle
+     *  @param address  The SocketAddress of the remote host
+     *  @param data     Buffer of data to send to the host
+     *  @param size     Size of the buffer in bytes
+     *  @param control     Storage for ancillary data
+     *  @param control_size   Size of  ancillary data
+     *  @return         Number of sent bytes on success, negative error
+     *                  code on failure
+     */
+    virtual nsapi_size_or_error_t socket_sendmsg(nsapi_socket_t handle, const SocketAddress &address,
```

Comment:
This isn't terribly close to POSIX `sendmsg`, it's an API halfway house between `sendto` and `sendmsg`. Maybe it should have a different name like `sendto_ancillary`?

Reply:
I agree, it's between sendto and sendmsg. I don't think we want a full blown `sendmsg`/`recvmsg` as it requires an extra buffer if there is more than one buffer submitted in the `iovec` field. Adding `ancillary` would talk to the one that are used to the POSIX API but for newcomers I always find it confusing. These are control flags, not just random additional/meta data. 

On the other hand I'm not sure what a good name would be.

Reply:
Yep, the iovec is nice-to-have occasionally, but implementing it as emulation on top of non-vector stuff is counter-productive. Most underlying stacks will not support it.

Maybe `_control` makes sense, as that is the actual name of the field in the `struct msghdr`.